### PR TITLE
configuration parameters for istio

### PR DESCRIPTION
**Pull request by d3vb0ox**

### CODE REVIEW

**Comment by smtm001** [comment]
Added a comment regarding code styling. LGTM functionality wise.

File: main.tf [hunk 1]
```diff
@@ -1,3 +1,23 @@
+data "aws_subnets" "istio_private" {
```
Comment (v47) [inline main.tf:1]: it's more reliable to have `istio_private_subnet_ids` and `istio_public_subnet_ids`.

File: main.tf [hunk 2]
```diff
@@ -15,7 +35,7 @@ resource "helm_release" "istio-discovery" {
   namespace        = var.istio_release_namespace
   create_namespace = true
   values           = [
-    templatefile("${path.module}/values-istiod.yaml", {})
+    templatefile("${path.module}/values/values-istiod.yaml", {})
```
Comment (v47) [inline main.tf:38]: you already have a folder named values, `values-*.yaml` is redundant. `/values/istiod.yaml` is expressive enough.

File: variables.tf [hunk 3]
```diff
@@ -1,3 +1,31 @@
+variable "vpc_id" {
+  type        = string
+  description = "The ID of the VPC."
+}
+
+variable "istio_external_gateway_lb_certs" {
+  type        = string
+  description = "The certificates for the Istio external gateway load balancer."
```
Comment (v47) [inline variables.tf:8]: is it an array or a comma-separated list?

File: variables.tf [hunk 4]
```diff
@@ -1,3 +1,31 @@
...
+}
+
+variable "istio_external_gateway_service_kind" {
+  type        = string
+  default     = "NodePort"
+  description = "The type of service for the Istio external gateway."
```
Comment (v47) [inline variables.tf:14]: can you add `validation` block?

File: variables.tf (hunk 3 above)
Comment (d3vb0ox) [reply variables.tf:33]: changed to list of string and joining with , on actual call

File: main.tf (hunk 2 above)
Comment (d3vb0ox) [reply main.tf]: removed values from file names

**Review by v47 — APPROVED** [review]

**Pull request merged**